### PR TITLE
Deprecate operator dispatch in `Var`

### PR DESCRIPTION
**Pull request by neNasko1**
Operator dispatch is a too high-level feature for `spox`. Also it is currently clunky to use from a user standpoint as it essentially "monkey-patches" the dispatcher in `spox`.  

A more appropriate way to access this behaviour is by using [ndonnx](https://github.com/Quantco/ndonnx).

# Checklist

- [X] Added a `CHANGELOG.rst` entry

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -18,6 +18,10 @@ Change log
 
 - Value propagation of string tensors no longer raises an erroneous ``ValueError`` in some instances.
 
+**Pending breaking changes**
+
+- Usage of ``operator_overloading`` now triggers a ``DeprecationWarning``. This API is scheduled for removal in Spox ``0.14.0``. Consider using ``ndonnx`` as an alternative.
```
Comment (cbourjau) [inline CHANGELOG.rst:23]:
```suggestion
- Importing :func:`spox._future.operator_overloading`` now triggers a ``DeprecationWarning``. The function will be removed in a future release. Consider using ``ndonnx`` as an alternative.
```

**Review by cbourjau — CHANGES_REQUESTED** [review]
I doesn't make a big difference, but technically we want to be sure that we warn users if they do anything that will break in the future. If we remove `operator_overloading` in the future any `from spox._future import operator_overloading` will already break regardless of usage. As such, we would ideally already warn on that level. One way to do so is to rename the function to `_operator_overloading` and then define a `getattr` function in the module to return it if a user imports "operator_overloading". We may then raise the warning from `getattr` rather than when the user actually calls the function. This comes with the added benefit that it would already remove the function from the documentation, too.

File: src/spox/_future.py [hunk 2]
```diff
@@ -201,6 +202,10 @@ def operator_overloading(
     ...    return x * y
     >>> assert foo()._get_value() == np.array(6)
     """
+    warnings.warn(
+        "using operator_overloading is deprecated, consider using https://github.com/Quantco/ndonnx instead",
```
[the remark this answers is not in the document]
Comment (neNasko1) [reply src/spox/_future.py]: Done!

**Comment by neNasko1** [comment]
> I doesn't make a big difference, but technically we want to be sure that we warn users if they do anything that will break in the future. If we remove `operator_overloading` in the future any `from spox._future import operator_overloading` will already break regardless of usage. As such, we would ideally already warn on that level. One way to do so is to rename the function to `_operator_overloading` and then define a `getattr` function in the module to return it if a user imports "operator_overloading". We may then raise the warning from `getattr` rather than when the user actually calls the function. This comes with the added benefit that it would already remove the function from the documentation, too.

I think it should now work as described here^

Note: These are how the different situations get handled after the change, using the [developer mode](https://docs.python.org/3/library/devmode.html#devmode):
```python
# warning
from spox._future import operator_overloading

# no warning
import spox._future as sf

# warning
sf.operator_overloading

# no warning
from spox._future import set_type_warning_level

# ImportError
from spox._future import non_existant_function
```

File: src/spox/_future.py [hunk 3]
```diff
@@ -221,5 +233,6 @@ def operator_overloading(
     "set_value_prop_backend",
     "value_prop_backend",
     # Operator overloading on Var
-    "operator_overloading",
+    "operator_overloading",  # noqa: F822
+    "__getattr__",
```
Comment (cbourjau) [inline src/spox/_future.py:237]:
```suggestion
```
`__all__` is only used for IDEs and when generating the docs. It has not impact on whether something can actually be imported. We don't want to keep deprecated things around in the docs, so it is fine to remove it here. We should also remove it from the actual docs [here](https://spox.readthedocs.io/en/latest/manual/unstable.html#operator-overloading) in a follow-up PR.

**Review by cbourjau — APPROVED** [review]

File: src/spox/_future.py (hunk 3 above)
Comment (neNasko1) [reply src/spox/_future.py]: > We should also remove it from the actual docs [here](https://spox.readthedocs.io/en/latest/manual/unstable.html#operator-overloading) in a follow-up PR.

Shouldn't this live in the PR removing the actual function?

File: CHANGELOG.rst [hunk 4]
```diff
@@ -18,6 +18,10 @@ Change log
 
 - Value propagation of string tensors no longer raises an erroneous ``ValueError`` in some instances.
 
+**Pending breaking changes**
```
Comment (cbourjau) [inline CHANGELOG.rst:21]:
```suggestion
**Deprecation**
```

**Pull request merged**